react/LocationSearch.tsx

```diff
@@ -1,37 +1,59 @@
-import React, { useState, useMemo } from 'react'
+import React, { useState, useMemo, useRef } from 'react'
 import { FormattedMessage } from 'react-intl'
-import { Input, IconSearch, IconClear } from 'vtex.styleguide'
+import { Input, IconSearch, IconClear, IconWarning } from 'vtex.styleguide'
+import { positionMatchWidth } from '@reach/popover'
+
 import {
   Combobox,
   ComboboxInput,
   ComboboxOption,
   ComboboxPopover,
   ComboboxList,
-} from '@reach/combobox'
-import '@reach/combobox/styles.css'
+} from './components/Combobox'
+import PlaceIcon from './components/PlaceIcon'
 
-import { addresses as mockedAddresses } from './addresses'
-import styles from './LocationSearch.css'
+interface Interval {
+  offset: number
+  size: number
+}
 
-const MAX_DROPDOWN_ADDRESSES = 6
+export interface Suggestion {
+  description: string
+  mainText: string
+  mainTextMatchInterval: Interval
+  secondaryText: string
+}
 
-// This function will be replaced in the future, after integrating the
-// component with GraphQL queries.
-const getAddresses = (searchTerm: string) => {
-  return mockedAddresses
-    .filter((address: string) =>
-      address.toLocaleLowerCase().includes(searchTerm.toLocaleLowerCase())
-    )
-    .slice(0, MAX_DROPDOWN_ADDRESSES)
+const renderSuggestionText = (suggestion: Suggestion) => {
+  const { mainText } = suggestion
+  const { offset, size } = suggestion.mainTextMatchInterval
+  return (
+    <div className="truncate c-muted-2">
+      <span className="c-on-base">{mainText.substr(0, offset)}</span>
+      <span className="c-on-base b">{mainText.substr(offset, size)}</span>
+      <span className="c-on-base">{mainText.substr(size + offset)}</span>
+      <span>{` ${suggestion.secondaryText}`}</span>
+    </div>
+  )
 }
```

Comment:
#### About the full address being displayed

In order to handle this, I’ve opted for implementing something similar to Google autocomplete ([see AutocompletePrediction Interface](https://developers.google.com/maps/documentation/javascript/reference/places-autocomplete-service#AutocompletePrediction)). When the user inputs the text "`bri`" into the input box, an answer like this will be returned:

```json
[
  {
    "description": "Cambridge, Rockford, Illinois",
    "mainText": "Cambridge",
    "secondaryText": "Rockford, Illinois",
    "mainTextMatchInterval": {
      "offset": 3,
      "size": 3
    }
  },
  {
    "description": "Briar Crest, Hagerstown, Maryland",
    "mainText": "Briar Crest",
    "secondaryText": "Hagerstown, Maryland",
    "mainTextMatchInterval": {
      "offset": 0,
      "size": 3
    }
  }
]
```

Which will be converted to (just the first one, for simplicity):

```html
<div className="truncate c-muted-2">
  <span className="c-on-base">Cam</span>
  <span className="c-on-base b">bri</span>
  <span className="c-on-base">dge</span>
  <span>Rockford, Illinois</span>
</div>
```

And displayed like:

<img src="https://user-images.githubusercontent.com/26108090/88711270-52e8be00-d0ee-11ea-8e50-f0f2a1016e19.png" width="300"/>

Explanation of the fields:

- `description`: the text that will go to the input box when the user selects it
- `mainText`: probably the street name (along with the number, maybe?)
- `secondaryText`: larger administrative regions like city, state and country
- `mainTextMatchInterval`: used to highlight the part of the `mainText` that matched the user input. This can be improved to hold multiple matches and/or include the `secondaryText`, but for now, I think imitating the Google behavior might be sufficient (which is just like that).

This isn't very flexible, but looks sufficient to me. Opinions?

Reply:
Seems enough for me as well.